.github/workflows/test.yaml

```
@@ -35,28 +35,34 @@ jobs:
         # We run this job multiple times with different parameterization
         # specified below, these parameters have no meaning on their own and
         # gain meaning on how job steps use them.
-        include:
-          - python: 3.5
-          - python: 3.6
-          - python: 3.7
-          - python: 3.8
-          - python: 3.9
+        jupyterlab_version: [1, 2, 3]
+        python: [3.6, 3.7, 3.8, 3.9]
 
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
         with:
           python-version: "${{ matrix.python }}"
-
+      - name: Cache wheel
+        id: cache-wheel
+        uses: actions/cache@v2
+        with:
+          path: ./dist/
+          key: ${{ runner.os }}-dist
+      - name: Build Python package
+        id: build-package
+        run: |
+          pip install jupyter_packaging wheel jupyterlab
+          python setup.py sdist bdist_wheel
       - name: Install Python dependencies
         # NOTE: See CONTRIBUTING.md for a local development setup that differs
         #       slightly from this.
         run: |
           pip install --upgrade pip
-          pip install .
+          pip install jupyter_packaging jupyterlab~=${{ matrix.jupyterlab_version }}.0
+          pip install ./dist/jupyter_server_proxy-3.0.0-py3-none-any.whl
```

This is supposed to be the current version rather than 3.0.0 in general right? I think the suggestion below will install all matching wheels. Due to caching logic I'm not 100% if this is an issue or not.

```suggestion
          pip install ./dist/jupyter_server_proxy-*.whl
```